### DISCUSSION TITLE
detektProjectBaseline slow?

### DISCUSSION THREAD

**Discussion by adrianlandborn** [detekt/detekt#6876] [Q&A]
Hi,
Were using detekt with a multimodule (about 10 modules) Android project. I am trying to move from separate baseline files to having only one in order to get the Android studio/IntelliJ plugin to work (only supports 1 baseline file). The new setup is working but it is really slow.
- detektBaseline: about **30 sec**
- detectProjectBaseline: about: **3 min**

ProjectBaseline in Kotlin
```
 val detektProjectBaseline by tasks.registering(DetektCreateBaselineTask::class) {
        description = "Overrides current baseline."
        parallel.set(true)
        buildUponDefaultConfig.set(false)
        ignoreFailures.set(false)
        setSource(files(rootDir))
        config.setFrom(files("$rootDir/config/detekt.yml"))
        baseline.set(file("$rootDir/config/baseline.xml"))
        include("**/*.kt")
        include("**/*.kts")
        exclude("**/resources/**")
        exclude("**/build/**")
    }
```

Baseline in Groovy (were not able to get this to run if moved to project level build.gradle.kts for some reason)
```
detekt {
    parallel = true
    buildUponDefaultConfig = false
    disableDefaultRuleSets = false
    debug = false
    ignoreFailures = false
    ignoredBuildTypes = ["release"]
    baseline = file("${rootDir}/config/baseline.xml")
    config.setFrom(files("${rootDir}/config/detekt.yml"))
    reports {
    }
}
```

Is this expected? Any input?
Thanks

**Comment by adrianlandborn** [comment]
Any ideas from anyone?
We are hesitant of merging the baseline files due to performance loss. However, it would be awesome to be able to use the detekt plugin in the IDE to highlight issues and be able to "quick-fix" them.

**Reply by cortinico** [reply]
> Any ideas from anyone?

Have you considered profiling your task using an included build? Do you have an idea of how many files you're actually processing?

> Is this expected? Any input?

3 mins to generate the baseline feels like too much. Still without a reproducer or more data is really hard to help you here

**Reply by adrianlandborn** [reply]
We have merged the baseline files into one now. From our testing, the build/run time is quite high for modules even though there is barely any code to process in those modules. So we ended up just removing detekt from those modules. We have 20 modules where 10 of them are only used as resource packages, which means almost no code, and the code we have there is rarely touched.

From my understanding detekt with a single baseline file seems to run sequentially and not in parallel even though there is a `parallel`parameter. Am I right?

**Reply by cortinico** [reply]
> From my understanding detekt with a single baseline file seems to run sequentially and not in parallel even though there is a `parallel`parameter. Am I right?

If you set `parallel`, we're going to use a Java's ExecutorService to run the analysis and parsing of the files (see here: https://github.com/detekt/detekt/blob/8a3eac595ef70a909b39130c7204c38885deaa0c/detekt-core/src/main/kotlin/io/gitlab/arturbosch/detekt/core/KtTreeCompiler.kt#L44-L54)

So even if you're a single module with a single baseline, we're going to run in parallel.
